engine/runtime/src/main/java/org/enso/interpreter/runtime/Module.java

```diff
@@ -198,6 +216,27 @@ public void setLiteralSource(String source) {
     setLiteralSource(Rope.apply(source), null);
   }
 
+  /**
+   * Attaches names of submodules that should be accessible from this module
+   *
+   * @param names fully qualified names of (potentially virtual) modules' names
+   */
+  public void setDirectSubmoduleNames(QualifiedName[] names) {
+    directSubmoduleNames = names;
+  }
+
+  /**
+   * Return an array of directly available modules of this one, if any.
+   *
+   * @return a non-null, possibly empty, array of fully qualified names of modules
+   */
+  public QualifiedName[] getDirectSubmoduleNames() {
+    if (directSubmoduleNames == null) {
+      return new QualifiedName[0];
+    }
+    return directSubmoduleNames;
```

Comment:
Beware the array is modifiable! Returning `Collections.unmodifiableList(Arrays.asList(directSubmoduleNames))` would be safer.